### ISSUE TITLE
[v4.2.0]"TypeError: Cannot read property 'length' of undefined"

### ISSUE THREAD

**Issue by killzoner** [bcaudan/jasmine-spec-reporter#222]
## Bug report

Check with "x":

- [x ] I have this issue with the latest release of [jasmine-spec-reporter](https://github.com/bcaudan/jasmine-spec-reporter/releases) 
- [x ] I have this issue with the latest release of [jasmine](https://github.com/jasmine/jasmine/releases) 
- [ ] I can reproduce this issue with one of the [examples](https://github.com/bcaudan/jasmine-spec-reporter/tree/master/examples)

**Current behavior**
TypeError: Cannot read property 'length' of undefined
**Expected behavior**
Display nicely there is no error

**Minimal reproduction of the problem with instructions**
Protractor test with no error result

Hi, i'm having a "TypeError: Cannot read property 'length' of undefined" since v4.2.0 upgrade. Tested with 4.1.1 fine. I don't really have a repro right now but i managed to get it working with simple and no effect on the reporter behaviour (see pull request https://github.com/bcaudan/jasmine-spec-reporter/pull/223). Should probably investigate further to see why the property was set before and not now (maybe just an effect or refactoring occured with v4.2.0?).

Thanks!

**Comment by brandonaaskov** [comment]
Thank you @killzoner for creating this issue. That made it very easy for me to fix my problem. We were seeing all our specs pass but our builds fail on our Jenkins box (AWS). I couldn't repro this locally, but rolling back to `4.1.1` solved it for me.

**Comment by johnjbarton** [comment]
I also encounter this; I think I caused this in #214.  

I believe the problem is a change in jasmine versions: if jasmineDone sends  RunDetails.failedExpectations as defined but empty we are good (new behavior), but if it is undefined then we failed. That explains why the `yarn test` works in pr #214 (jasmine 2.7.0). My use cases fail with jasmine-spec-reporter 4.20 + jasmine 2.4.1 but pass with jasmine 2.7.0.

I think the fix in #223 is correct.
